test/internal/options.go

```diff
@@ -0,0 +1,263 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	dwsv1alpha1 "github.com/HewlettPackard/dws/api/v1alpha1"
+	"github.com/HewlettPackard/dws/utils/dwdparse"
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	lusv1alpha1 "github.com/NearNodeFlash/lustre-fs-operator/api/v1alpha1"
+	nnfv1alpha1 "github.com/NearNodeFlash/nnf-sos/api/v1alpha1"
+)
+
+// TOptions let you configure things prior to a test running or during test
+// execution. Nil values represent no configuration of that type.
+type TOptions struct {
+	stopAfter         *TStopAfter
+	storageProfile    *TStorageProfile
+	persistentLustre  *TPersistentLustre
+	globalLustre      *TGlobalLustre
+	cleanupPersistent *TCleanupPersistentInstance
+}
+
+type TStopAfter struct {
+	state dwsv1alpha1.WorkflowState
+}
+
+// Stop after lets you stop a test after a given state is reached
+func (t *T) StopAfter(state dwsv1alpha1.WorkflowState) *T {
+	t.options.stopAfter = &TStopAfter{state: state}
+	return t
+}
+
+func (t *T) ShouldTeardown() bool {
+	return t.options.stopAfter == nil
```

Comment:
I think you just described the error case. That is what I'd expect wlm to do for an error. A test suite that brings the workflow to all the states, stops, then jumps to teardown is needed to make sure teardown can cleanup from any state for any workflow.

Reply:
It might be useful to jump to teardown from any state to make sure things transition and clean up properly. Perhaps a new option like `SkipToTeardownAfter(state)`. That's a little more nuanced than what I'm setting out to do though.